packages/core/admin/admin/src/components/NpsSurvey/index.js

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+
+import { Flex, Button, Typography } from '@strapi/design-system';
+import { usePersistentState } from '@strapi/helper-plugin';
+
+const delays = {
+  postResponse: 90 * 24 * 60 * 60 * 1000, // 90 days in ms
+  postFirstDismissal: 7 * 24 * 60 * 60 * 1000, // 7 days in ms
+  postSubsequentDismissal: 90 * 24 * 60 * 60 * 1000, // 90 days in ms
+};
+
+const checkIfShouldShowSurvey = (settings) => {
+  const { enabled, lastResponseDate, firstDismissalDate, lastDismissalDate } = settings;
+
+  if (!enabled) {
+    return false;
+  }
+
+  if (lastResponseDate) {
+    // If the user has already responded to the survey
+    const timeSinceLastResponse = Date.now() - new Date(lastResponseDate).getTime();
+
+    if (timeSinceLastResponse >= delays.postResponse) {
+      return true;
+    }
+
+    return false;
+  }
+
+  if (lastDismissalDate) {
+    // If the user has dismissed the survey twice or more before
+    const timeSinceLastDismissal = Date.now() - new Date(lastDismissalDate).getTime();
+
+    if (timeSinceLastDismissal >= delays.postSubsequentDismissal) {
+      return true;
+    }
+
+    return false;
+  }
+
+  if (firstDismissalDate) {
+    // If the user has dismissed the survey before
+    const timeSinceFirstDismissal = Date.now() - new Date(firstDismissalDate).getTime();
+
+    if (timeSinceFirstDismissal >= delays.postFirstDismissal) {
+      return true;
+    }
+
+    return false;
+  }
+
+  // If the user has not interacted with the survey before
+  return true;
+};
```

Comment:
```suggestion
const checkIfShouldShowSurvey = (settings) => {
  const { enabled, lastResponseDate, firstDismissalDate, lastDismissalDate } = settings;

  if (!enabled) {
    return false;
  }

  if (lastResponseDate) {
    // If the user has already responded to the survey
    const timeSinceLastResponse = Date.now() - new Date(lastResponseDate).getTime();

    return timeSinceLastResponse >= delays.postResponse;
  }

  if (lastDismissalDate) {
    // If the user has dismissed the survey twice or more before
    const timeSinceLastDismissal = Date.now() - new Date(lastDismissalDate).getTime();

    return timeSinceLastDismissal >= delays.postSubsequentDismissal;
  }

  if (firstDismissalDate) {
    // If the user has dismissed the survey before
    const timeSinceFirstDismissal = Date.now() - new Date(firstDismissalDate).getTime();

    return timeSinceFirstDismissal >= delays.postFirstDismissal;
  }

  // If the user has not interacted with the survey before
  return true;
};
```

I'm still not sure why the ifs are in the order they are (you probably know why), I think for future devs it'd be super helpful just to write a big ol' conceptual comment above just explaining why it's written in this order etc. so it's clear to anyone else 👍🏼

Reply:
I added a big comment above, lmk if it helps